### PR TITLE
Vignette filter, tests, and test images

### LINKED ISSUE #24: Code review of vignette function

### PR DESCRIPTION
**Pull request by brendoncampbell**

### CODE REVIEW

File: pymagine/vignette_filter.py [hunk 1]
```diff
@@ -15,12 +15,67 @@ def vignette_filter(image, a=1.0):
...
+  
   Returns
   -------
   image
     image returned with the desired filter applied
-  """
\ No newline at end of file
+  """
+  if not isinstance(image_path, str):
+        raise TypeError("Image file path must be a string.")
```
Comment (Sukriti1312) [inline pymagine/vignette_filter.py]: Please correct the indentation

File: pymagine/vignette_filter.py [hunk 2]
```diff
@@ -15,12 +15,67 @@ def vignette_filter(image, a=1.0):
...
+    image_modified = np.copy(image)
+    image_modified[:,:,:] = 0
+    
+    # apply filter to each layer of the input image
+    image_modified[:,:,0] = image[:,:,0]*filt_2d_scaled
+    image_modified[:,:,1] = image[:,:,1]*filt_2d_scaled
+    image_modified[:,:,2] = image[:,:,2]*filt_2d_scaled
+    
+    # likely need to update below to return an image, rather than writing to disk
+    return cv2.imwrite('vignette.jpg',image_modified)
```
Comment (Sukriti1312) [inline pymagine/vignette_filter.py]: Please this code chunk indent to the left

**Review by Sukriti1312 — CHANGES_REQUESTED** [review]
Please make changes in the indentation. Thank you!

File: pymagine/vignette_filter.py (hunk 2 above)
Comment (brendoncampbell) [reply pymagine/vignette_filter.py]: @Sukriti1312 should be fixed in the latest commit

File: pymagine/vignette_filter.py [hunk 3]
```diff
@@ -4,23 +4,74 @@
...
+    Default: 0.5
+
+  y: float
+    Parameter for the centre point of the effect
+    along the y axis.
+    Default: 0.5
+  
+  """
+  if not isinstance(image_path, str):
+        raise TypeError("Image file path must be a string.")
```
Comment (Sukriti1312) [inline pymagine/vignette_filter.py]: This code line also needs to be intended to the left.
```suggestion
    raise TypeError("Image file path must be a string.")
```

**Review by Sukriti1312 — CHANGES_REQUESTED** [review]
One more code line needs to be correctly indented. Thank you!

File: pymagine/vignette_filter.py [hunk 4]
```diff
@@ -4,23 +4,79 @@
...
+    along the y axis.
+    Default: 0.5
+  
+  """
+  if not isinstance(image_path, str):
+<<<<<<< HEAD
+      raise TypeError("Image file path must be a string.")
+=======
+    raise TypeError("Image file path must be a string.")
+>>>>>>> b4a1b0dda89227aa7f0c3883276633ea1eb559f3
```
Comment (katieb1) [inline pymagine/vignette_filter.py]: merge conflict
```suggestion
      raise TypeError("Image file path must be a string.")
```

**Review by katieb1 — CHANGES_REQUESTED** [review]

**Review by katieb1 — APPROVED** [review]

**Review by Sukriti1312 — APPROVED** [review]
Looks good

**Pull request merged**